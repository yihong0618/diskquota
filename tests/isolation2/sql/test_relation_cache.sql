CREATE DATABASE tempdb1;
CREATE DATABASE tempdb2;

-- perpare extension
1:@db_name tempdb1: CREATE EXTENSION diskquota;
1:@db_name tempdb1: SELECT diskquota.wait_for_worker_new_epoch();
2:@db_name tempdb2: CREATE EXTENSION diskquota;
2:@db_name tempdb2: SELECT diskquota.wait_for_worker_new_epoch();

-- create a table in tempdb1
1:@db_name tempdb1: BEGIN;
1:@db_name tempdb1: CREATE TABLE t(i int);
1:@db_name tempdb1: INSERT INTO t select generate_series(1, 10000);

-- query relation_cache in tempdb2
2:@db_name tempdb2: SELECT count(*) from diskquota.show_relation_cache();

1:@db_name tempdb1: ABORT;

1:@db_name tempdb1: SELECT diskquota.pause();
1:@db_name tempdb1: SELECT diskquota.wait_for_worker_new_epoch();
1:@db_name tempdb1: DROP EXTENSION diskquota;
2:@db_name tempdb2: SELECT diskquota.pause();
2:@db_name tempdb2: SELECT diskquota.wait_for_worker_new_epoch();
2:@db_name tempdb2: DROP EXTENSION diskquota;
1q:
2q:

DROP DATABASE tempdb1;
DROP DATABASE tempdb2;
