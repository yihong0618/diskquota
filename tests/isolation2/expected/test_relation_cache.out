CREATE DATABASE tempdb1;
CREATE
CREATE DATABASE tempdb2;
CREATE

-- perpare extension
1:@db_name tempdb1: CREATE EXTENSION diskquota;
CREATE
1:@db_name tempdb1: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)
2:@db_name tempdb2: CREATE EXTENSION diskquota;
CREATE
2:@db_name tempdb2: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)

-- create a table in tempdb1
1:@db_name tempdb1: BEGIN;
BEGIN
1:@db_name tempdb1: CREATE TABLE t(i int);
CREATE
1:@db_name tempdb1: INSERT INTO t select generate_series(1, 10000);
INSERT 10000

-- query relation_cache in tempdb2
2:@db_name tempdb2: SELECT count(*) from diskquota.show_relation_cache();
 count 
-------
 0     
(1 row)

1:@db_name tempdb1: ABORT;
ABORT

1:@db_name tempdb1: SELECT diskquota.pause();
 pause 
-------
       
(1 row)
1:@db_name tempdb1: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)
1:@db_name tempdb1: DROP EXTENSION diskquota;
DROP
2:@db_name tempdb2: SELECT diskquota.pause();
 pause 
-------
       
(1 row)
2:@db_name tempdb2: SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t                         
(1 row)
2:@db_name tempdb2: DROP EXTENSION diskquota;
DROP
1q: ... <quitting>
2q: ... <quitting>

DROP DATABASE tempdb1;
DROP
DROP DATABASE tempdb2;
DROP
